tests/main.rs

```diff
@@ -210,6 +213,108 @@ fn it_connects_tls_both() {
     .test(|_| {})
 }
 
+#[test]
+#[cfg(feature = "tls")]
+fn it_connects_tls_both_with_delayed_server_read() {
```

Comment:
Would you mind leaving a comment here explaining why this test is valuable so that down the road when we run into it again, we recall what it's actually testing for?